sdk/storage/azure-storage-queue/src/main/java/com/azure/storage/queue/QueueConfiguration.java

```diff
@@ -2,11 +2,46 @@
 // Licensed under the MIT License.
 package com.azure.storage.queue;
 
+import com.azure.core.util.logging.ClientLogger;
+import java.io.IOException;
+import java.util.Properties;
+
 /*
  * Gets the SDK information for this library component.
  */
 class QueueConfiguration {
     //TODO: Eventually remove these hardcoded strings with https://github.com/Azure/azure-sdk-for-java/issues/3141
-    static final String NAME = "azure-storage-queue";
-    static final String VERSION = "12.0.0-preview.3";
+    private final ClientLogger logger = new ClientLogger(QueueClient.class);
+    private final Properties properties = new Properties();
+
+    public QueueConfiguration() {
+        loadProperties();
+    }
+
+    private void loadProperties() {
+        try {
+            properties.load(this.getClass().getClassLoader().getResourceAsStream("client.properties"));
```

Comment:
Instead of loading properties should we being using reflections and getting the package information?